### ISSUE TITLE
fstream include removed while required

### ISSUE THREAD

**Issue by vsapsai** [include-what-you-use/include-what-you-use#90]
Originally reported on Google Code with ID 90

```
What is the expected output? What do you see instead?

I run include-what-you-use on a large source file containing Google Test test cases.
On this particular file, there is a class extending 'testing::Test', which has protected
members of type 'ifstream'. As a result, an include is needed: '#include <fstream>'

However, IWYU suggests removing the '<fstream>' include, and I can't find out why.
I've created a smaller test case.


What version of the product are you using? On what operating system?
I'm afraid I'm using an older IWYU version (revision 352), since I had issues compiling
the newer version.
I'm running Ubuntu Linux 64-bit.

Please provide any additional information below.


```

Reported by `mathias.demare` on 2013-01-04 16:03:30

<hr>
- _Attachment: [foo.cpp](https://storage.googleapis.com/google-code-attachments/include-what-you-use/issue-90/comment-0/foo.cpp)_

**Comment by firewave** [comment]
```cpp
#include <fstream>
#include <iosfwd>

int main(void)
{
    std::ifstream istr;
}
```

```
input.cpp should add these lines:

input.cpp should remove these lines:
- #include <fstream>  // lines 1-1

The full include-list for input.cpp:
#include <iosfwd>  // for ifstream
---
```

If you remove that include it will no longer compile.

**Comment by bsergean** [comment]
I ran into that problem too, with:

```
include-what-you-use 0.20 based on Homebrew clang version 16.0.6
```

**Comment by firewave** [comment]
It appears this have been resolved with 0.21. Using `include-what-you-use-0.21-1.1` on OpenSUSE Tumbleweed I was able to remove all the `IWYU pragma: keep` for `sstream` and `fstream` includes.

Unfortunately I do not have that version on a local installation so I cannot completely verify this.

**Comment by kimgr** [comment]
Phew! I finally built some infrastructure to be able to bisect where this bug disappeared.

It turns out it was silently fixed in https://github.com/include-what-you-use/include-what-you-use/commit/bc5221b570f4c70eb2601a79c79da2832a1c7aac, released as part of IWYU 0.21.

Thanks @bolshakov-a for taking care of this!

**Comment by bolshakov-a** [comment]
Wow, interesting... Glad to see.
